### PR TITLE
mgr/dashboard: fix average object size calculation in rgw overview dashboard

### PR DESCRIPTION
**Pull request by aaSharma14**
Consider just the data pools for average object size calculation

Fixes: https://tracker.ceph.com/issues/62318

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/services/cluster.py [hunk 1]
```diff
@@ -50,25 +50,45 @@ def get_capacity(cls) -> ClusterCapacity:
...
-                rgw_pool_names.append(pool['pool_name'])
-        return rgw_pool_names
+                name = pool['pool_name']
+                size = pool['size']
+                rgw_pool_names.append(name)
+                rgw_pool_size[name] = size
+
+        rgw_pools_data.append(rgw_pool_names)
+        rgw_pools_data.append(rgw_pool_size)
+        return rgw_pools_data
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/services/cluster.py:94]:
```
    def get_rgw_pools(cls):
        rgw_pool_size = {}

        osd_map = mgr.get('osd_map')
        for pool in osd_map['pools']:
            if 'rgw' in pool.get('application_metadata', {}):
                name = pool['pool_name']
                rgw_pool_size[name] = pool['size']
        return rgw_pool_size
```

File: src/pybind/mgr/dashboard/services/cluster.py [hunk 2]
```diff
@@ -50,25 +50,45 @@ def get_capacity(cls) -> ClusterCapacity:
...
-                total_pool_bytes_used = total_pool_bytes_used + pool['stats']['bytes_used']
-        if total_objects != 0:
-            average_object_size = total_pool_bytes_used / total_objects
-        return ClusterCapacity(total_avail_bytes=df['stats']['total_avail_bytes'],
-                               total_bytes=df['stats']['total_bytes'],
-                               total_used_raw_bytes=df['stats']['total_used_raw_bytes'],
-                               total_objects=total_objects,
-                               total_pool_bytes_used=total_pool_bytes_used,
-                               average_object_size=average_object_size)._asdict()
+            if pool['name'] in rgw_pools_data[0]:
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/services/cluster.py:58]:
```suggestion
            if pool['name'] in rgw_pools_data:
```

File: src/pybind/mgr/dashboard/services/cluster.py (hunk 1 above)
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/services/cluster.py]: Done, thanks @cloudbehl

**Review by cloudbehl — APPROVED** [review]
Lgtm

**Comment by nizamial09** [comment]
e2e failures [are unrelated](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/13288/consoleFull#-180209496973a8703b-5adb-41c5-84a0-8cf4e065ba3d)

**Pull request merged**